src/CodeGen_LLVM.cpp

```diff
@@ -1466,10 +1466,28 @@ void CodeGen_LLVM::codegen(const Stmt &s) {
     value = nullptr;
     s.accept(this);
 }
+namespace {
+
+bool is_power_of_two(int x) {
+    return (x & (x - 1)) == 0;
+}
+
+int next_power_of_two(int x) {
+    for (int p2 = 1;; p2 *= 2) {
+        if (p2 >= x) {
```

Comment:
I think this could theoretically loop forever... the max value of int is not a power of 2, so it could overflow and wrap around and keep trying.

Reply:
Good point, replaced it with the one which ends.